python-checks/src/main/java/org/sonar/python/checks/tests/TestShouldBeSkippedExplicitlyCheck.java

```diff
@@ -0,0 +1,129 @@
+    "assertNotIsInstance", "assertRaises", "assertRaisesRegexp", "assertRaisesRegex", "assertAlmostEqual", "assertNotAlmostEqual",
+    "assertGreater", "assertGreaterEqual", "assertLess", "assertLessEqual", "assertRegexpMatches", "assertNotRegexpMatches",
+    "assertItemsEqual", "assertDictContainsSubset"));
+  private static final Tree.Kind[] literalsKind = {Tree.Kind.STRING_LITERAL, Tree.Kind.NUMERIC_LITERAL, Tree.Kind.LIST_LITERAL,
+    Tree.Kind.BOOLEAN_LITERAL_PATTERN, Tree.Kind.NUMERIC_LITERAL_PATTERN, Tree.Kind.NONE_LITERAL_PATTERN, Tree.Kind.STRING_LITERAL_PATTERN,
+    Tree.Kind.SET_LITERAL, Tree.Kind.DICTIONARY_LITERAL};
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, ctx -> {
+      FunctionDef functionDef = (FunctionDef) ctx.syntaxNode();
+
+      if (!functionDef.name().name().startsWith("test")) {
+        return;
+      }
+
+      ReturnStatement returnStatement = getReturnStatementInIfFirstStatement(functionDef);
+      if (returnStatement == null) {
+        return;
+      }
+
+      // check there is an assert like statement in the function
+      if (isOrContainAssertStatement(functionDef)) {
+        ctx.addIssue(returnStatement, MESSAGE);
+      }
+    });
+  }
+
+  private static ReturnStatement getReturnStatementInIfFirstStatement(FunctionDef functionDef) {
+    // check first non assignment statement is an if with a return
+    for (Statement statement : functionDef.body().statements()) {
+      // skip literal assignment expression
+      if (statement.is(Tree.Kind.ASSIGNMENT_STMT) && ((AssignmentStatement) statement).assignedValue().is(literalsKind)) {
+        continue;
+      }
+
+      if (statement.is(Tree.Kind.IF_STMT)) {
+        IfStatement ifStatement = (IfStatement) statement;
+        List<Statement> statements = ifStatement.body().statements();
+        if (statements.get(0).is(Tree.Kind.RETURN_STMT)) {
+          return (ReturnStatement) statements.get(0);
+        }
+      }
+
+      return null;
+    }
+    return null;
+  }
+
+  private static boolean isOrContainAssertStatement(Tree node) {
+    AssertOrCallAssertVisitor assertOrCallAssertVisitor = new AssertOrCallAssertVisitor();
+    node.accept(assertOrCallAssertVisitor);
+    return assertOrCallAssertVisitor.hasACallToAssert || assertOrCallAssertVisitor.hasANativeAssert;
+  }
+
+  static class AssertOrCallAssertVisitor extends BaseTreeVisitor {
+    boolean hasANativeAssert = false;
+    boolean hasACallToAssert = false;
+
+    @Override
+    public void visitAssertStatement(AssertStatement pyAssertStatementTree) {
+      hasANativeAssert = true;
+      super.visitAssertStatement(pyAssertStatementTree);
+    }
+
+    @Override
+    public void visitQualifiedExpression(QualifiedExpression pyQualifiedExpressionTree) {
+      Optional<String> name = tryGetName(pyQualifiedExpressionTree.qualifier());
```

Comment:
This could probably be made even simpler by checking if `pyQualifiedExpressionTree.qualifier().is(Tree.Kind.NAME)` and then just casting it and accessing its `name()` directly. That way, you don't need a different method or optional, etc (this is a very common pattern we use).
Maybe irrelevant if you end up using the callee symbol instead.

Reply:
Thanks for confirming, I was afraid that it would be a bit to bulky to put directly in the IF condition.
I adapted.